core/src/main/java/com/segment/analytics/kotlin/core/Analytics.kt

```diff
@@ -670,6 +670,41 @@ open class Analytics protected constructor(
      * - Returns: A string representing the version in "BREAKING.FEATURE.FIX" format.
      */
     fun version() = Analytics.version()
+
+    /**
+     * Provides a list of finished, but unsent events.
+     */
+    fun pendingUploads(): List<String> {
+        return parseFilePaths(storage.read(Storage.Constants.Events))
+    }
+
+    /**
+     * Purge all pending event upload files.
+     */
+    fun purgeStorage() {
+        // do not call purgeStorage(filePath: String)
+        // since we want to schedule all files deletion in the same dispatch all at once
+        analyticsScope.launch(fileIODispatcher) {
+            for (file in pendingUploads()) {
+                try {
+                    storage.removeFile(file)
```

Comment:
Is there maybe a way to remove the folder of pending upload files, instead of each file?

Reply:
java.io might have an api to delete a collection. but for us to implement it, we will have to add api to `EventFileManager`, `Storage` and all the implementations of `Storage`. not worth the effort, since the java.oi api eventually delete files one by one